### ISSUE TITLE
VL-CMU-CD aligned viewpoints

### ISSUE THREAD

**Issue by Chumsy0725** [ChadLin9596/Robust-Scene-Change-Detection#4]
Hello Lin, 

Thanks for releasing the codebase. I am trying to reproduce the results mentioned in T1. I trained the model on VL-CMU-CD with the default parameters in your train.yaml file (for 100 epochs). I am getting 77.4 vs 79.5 in T1 for aligned viewpoints. Any insight about what may cause this discrepancy? 

Thanks

![Image](https://github.com/user-attachments/assets/19f4d80d-9bd5-43f8-b3c5-3bf4e24ace8a)

**Comment by ChadLin9596** [comment]
Hi Chamuditha,

I notice that your epoch is 99. Did you evaluate using `last.pth`?
Typically, I evaluate using `best.val.pth` since it is selected based on the best validation performance. The best result can sometimes appear before the final epoch.

Additionally, Different Viewpoint Augmentation is enabled by default in `train.yaml` (line-64). So a fairer comparsion, you should refer to T4 (your 0.774 vs my 0.787). If you want to compare with T1, please modify `line 64` to `diff-augment: False`. 

That said, my training code does not guarantee 100% reproducibility of results, even though I explicitly fix the random seed using `utils_torch.seed_everything(_seed)`. Checkpoints trained under the same hardware and environment can still exhibit slight variations in performance.

If you have any insights on improving reproducibility, feel free to submit a pull request. I’d really appreciate your contributions!

Best,

**Comment by Chumsy0725** [comment]
Hi Lin, 

Thanks for your extensive response. I understand that it is not always possible to reproduce the results exactly as they are - I just wanted to check with you to see if I'm getting the best possible results. 

By changing the `diff-augmente=False` I was able to reproduce the results. 

Thanks again, Best,
